File: backends/php-laravel/app/UseCases/JwtHistory/JwtHistoryService.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\UseCases\JwtHistory;

use App\Domain\Client\Condition\JwtHistoryCondition;
use App\Domain\Client\Repositories\JwtHistoryRepository;
use App\Domain\Client\ValueObjects\JwtHistoryListVo;
use App\Support\Enums\SortType;
use App\Support\Repositories\Conditions\Option;
use App\Support\Services\AbstractService;
use App\UseCases\JwtHistory\Dtos\JwtHistoryDto;

/**
 * JWT履歴Serviceクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\UseCases\JwtHistory
 */
class JwtHistoryService extends AbstractService
{
    /**
     * コンストラクタ。
     *
     * @param JwtHistoryRepository $repository JWT履歴Repository
     */
    public function __construct(
        private readonly JwtHistoryRepository $repository,
    ) {
    }

    /**
     * クライアントに紐づくJWT履歴一覧を返します。
     *
     * @param JwtHistoryDto $dto JWT履歴DTO
     * @return JwtHistoryListVo JWT履歴一覧ValueObject
     */
    public function getHistories(JwtHistoryDto $dto): JwtHistoryListVo
    {
        if (empty($dto->sort)) {
            $dto->sort = 'issue_at';
        }
        if ($dto->sortType === SortType::NONE) {
            $dto->sortType = SortType::DESC;
        }

        $condition = new JwtHistoryCondition();
        $condition->clientId = $dto->clientId;
        $condition->option = new Option($dto->offset, $dto->limit, $dto->sort, $dto->sortType);

        $count = $this->repository->countByClientId($condition);
        $list = $this->repository->findByClientId($condition);

        $vo = new JwtHistoryListVo();
        $vo->assignItems($list);
        $vo->setCount($count);
        $vo->setPaging($dto->offset, $dto->limit, $dto->sort, $dto->sortType);

        return $vo;
    }
}

File: backends/java-spring/src/main/java/com/authorization/usecases/jwthistory/JwtHistoryService.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.usecases.jwthistory;

import com.authorization.domain.client.condition.JwtHistoryCondition;
import com.authorization.domain.client.entities.JwtHistory;
import com.authorization.domain.client.repositories.JwtHistoryRepository;
import com.authorization.domain.client.valueobjects.JwtHistoryListVo;
import com.authorization.support.enums.SortType;
import com.authorization.support.repositories.conditions.Option;
import com.authorization.support.services.AbstractService;
import com.authorization.usecases.jwthistory.dtos.JwtHistoryDto;
import java.util.List;

/**
 * JWT履歴Serviceクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
public class JwtHistoryService extends AbstractService {

    private final JwtHistoryRepository repository;

    /**
     * コンストラクタ。
     *
     * @param repository JWT履歴Repository
     */
    public JwtHistoryService(JwtHistoryRepository repository) {
        this.repository = repository;
    }

    /**
     * クライアントに紐づくJWT履歴一覧を返します。
     *
     * @param dto JWT履歴DTO
     * @return JWT履歴一覧ValueObject
     */
    public JwtHistoryListVo getHistories(JwtHistoryDto dto) {
        if (dto.getSort() == null || dto.getSort().isEmpty()) {
            dto.setSort("issue_at");
        }
        if (dto.getSortType() == SortType.NONE) {
            dto.setSortType(SortType.DESC);
        }

        JwtHistoryCondition condition = new JwtHistoryCondition();
        condition.setClientId(dto.getClientId());
        condition.setOption(new Option(dto.getOffset(), dto.getLimit(), dto.getSort(), dto.getSortType()));

        int count = repository.countByClientId(condition);
        List<JwtHistory> list = repository.findByClientId(condition);

        JwtHistoryListVo vo = new JwtHistoryListVo();
        vo.setItems(list);
        vo.setCount(count);
        vo.setOffset(dto.getOffset());
        vo.setLimit(dto.getLimit());
        vo.setSort(dto.getSort());
        vo.setSortType(dto.getSortType());
        return vo;
    }
}
